cvat-canvas/src/typescript/drawHandler.ts

```diff
@@ -121,22 +123,35 @@ export class DrawHandlerImpl implements DrawHandler {
 
         const findIntersectionsWithFrameBorders = (p1: Point, p2: Point, direction: Direction): number[] => {
             const resultPoints = [];
+            const leftLine = [
+                { x: 0, y: 0 },
+                { x: 0, y: frameHeight },
+            ];
+            const topLine = [
+                { x: frameWidth, y: 0 },
+                { x: 0, y: 0 },
+            ];
+            const rightLine = [
+                { x: frameWidth, y: frameHeight },
+                { x: frameWidth, y: 0 },
+            ];
+            const bottomLine = [
+                { x: 0, y: frameHeight },
+                { x: frameWidth, y: frameHeight },
+            ];
+
             if (direction === Direction.Horizontal) {
-                resultPoints.push(...findInersection(p1, p2, { x: 0, y: 0 }, { x: 0, y: frameHeight }));
-                resultPoints.push(
-                    ...findInersection(p1, p2, { x: frameWidth, y: frameHeight }, { x: frameWidth, y: 0 }),
-                );
+                resultPoints.push(...findInersection(p1, p2, leftLine[0], leftLine[1]));
+                resultPoints.push(...findInersection(p1, p2, rightLine[0], rightLine[1]));
             } else {
-                resultPoints.push(
-                    ...findInersection(p1, p2, { x: 0, y: frameHeight }, { x: frameWidth, y: frameHeight }),
-                );
-                resultPoints.push(...findInersection(p1, p2, { x: frameWidth, y: 0 }, { x: 0, y: 0 }));
+                resultPoints.push(...findInersection(p1, p2, bottomLine[0], bottomLine[1]));
+                resultPoints.push(...findInersection(p1, p2, topLine[0], topLine[1]));
             }
 
             if (resultPoints.length === 4) {
                 if (
-                    Math.sign(resultPoints[0] - resultPoints[2]) !== Math.sign(p1.x - p2.x)
-                    && Math.sign(resultPoints[1] - resultPoints[3]) !== Math.sign(p1.y - p2.y)
+                    (p1.x === p2.x || Math.sign(resultPoints[0] - resultPoints[2]) !== Math.sign(p1.x - p2.x))
+                    && (p1.y === p2.y || Math.sign(resultPoints[1] - resultPoints[3]) !== Math.sign(p1.y - p2.y))
```

Comment:
@azhavoro  @ActiveChooN 

The core change to fix (others are to satisfy eslint, minor refactoring, and some minor bug fixes). 

As I understand the issue was following:
After the first (Horizontal) crop p1.x in a corner case could be equal to p2.x).
The same if we reverse crop order (first is Vertical), then p1.y in a corner case could be equal to p2.y.

In these corner cases the condition can be false and we do not reverse points when it is actually necessary.